Plugins/Creature/Creature.cpp

```diff
@@ -3235,3 +3237,79 @@ NWNX_EXPORT ArgumentStack OverrideRangedProjectileVFX(ArgumentStack&& args)
 
     return {};
 }
+static void InitInitiativeRollHook()
+{
+    static Hooks::Hook pResolveInitiative_hook = Hooks::HookFunction(Functions::_ZN12CNWSCreature17ResolveInitiativeEv,
+    (void *) +[](CNWSCreature *pCreature) -> void
+    {
+        auto initMod = pCreature->nwnxGet<int32_t>("INITIATIVE_MOD").value_or(0);
+        if (!initMod)
+            return pResolveInitiative_hook->CallOriginal<void>(pCreature);
+
+        if (pCreature->m_bInitiativeExpired == 1)
+        {
+            auto pStats = pCreature->m_pStats;
+            auto diceRoll = Globals::Rules()->RollDice(1, 20);
+            auto mod = pStats->GetDEXMod(0);
+            if (pStats->HasFeat(Constants::Feat::EpicSuperiorInitiative))
+                mod += Globals::Rules()->GetRulesetIntEntry("EPIC_SUPERIOR_INITIATIVE_BONUS", 8);
+            else if (pStats->HasFeat(Constants::Feat::ImprovedInitiative))
+                mod += Globals::Rules()->GetRulesetIntEntry("IMPROVED_INITIATIVE_BONUS", 4);
+            if (pStats->HasFeat(Constants::Feat::Blooded))
+                mod += Globals::Rules()->GetRulesetIntEntry("BLOODED_INITIATIVE_BONUS", 2);
+            if (pStats->HasFeat(Constants::Feat::Thug))
+                mod += Globals::Rules()->GetRulesetIntEntry("THUG_INITIATIVE_BONUS", 2);
+
+            // Add creature bonus
+            mod += initMod;
+
+            pCreature->m_nInitiativeRoll = diceRoll + mod;
+            auto *pPlayer = Globals::AppManager()->m_pServerExoApp->GetClientObjectByObjectId(
+                    pCreature->m_idSelf);
+            if (pPlayer)
+            {
+                auto *pData = new CNWCCMessageData;
+                pData->SetInteger(0, diceRoll);
+                pData->SetInteger(1, mod);
+                pData->SetObjectID(0, pCreature->m_idSelf);
+                auto *pMessage = static_cast<CNWSMessage *>(Globals::AppManager()->m_pServerExoApp->GetNWSMessage());
+                if (pMessage)
+                {
+                    pMessage->SendServerToPlayerCCMessage(pPlayer->m_nPlayerID,
+                                                          Constants::MessageClientSideMsgMinor::Initiative,
+                                                          pData, nullptr);
+                }
```

Comment:
Need to `delete pData;` here. Or maybe you could just allocate it on the stack..

Reply:
Thanks clippy, caught that in Race as well and corrected with this change.